util/check_tool_requirements.py

```diff
@@ -96,8 +124,13 @@ def pip3_get_version(tool):
     return None
 
 
-def check_version(requirements, tool_name, getter):
+def check_version(requirements, tool_name, getter, version_converter = None):
```

Comment:
I think we're converting Verible versions twice: once here and once in `get_verible_version`. Maybe we can just do the latter, and drop this hunk?

If not, there's a nit here: I think the usual format is no spaces around the '=' sign for default arguments. Oddly, flake8 doesn't seem to be complaining about this line: I'm not sure why.

Reply:
Yeah I know this is not very nice. But the reason I did this twice is that we expose the full version string from `tool_requirements.py` in the documentation as part of the installation instructions: https://docs.opentitan.org/doc/ug/install_instructions/

Looks like I adhered to our SV style guide here ;) let me change that.

Reply:
Oh, I see! For anyone else following along, the point is that we store the required verible version in `tool_requirements.py` unmangled, partly because it's less confusing and partly because it gets copied from there into our documentation. But then we need to compare that version with the installed one. To do so, we convert both versions into "semver" format and then compare the results. The conversion of the version that's currently installed happens in `get_verible_version`; the conversion of the required version happens here.

Phew. However, maybe we should convert both versions in `check_version` (and remove the conversion from `get_verible_version`)? That feels slightly more contained to me.

Reply:
Agreed, I've rearranged this such that the two conversions are called side-by-side in `check_version`.